src/pint/observatory/topo_obs.py

```diff
@@ -147,39 +168,39 @@ def __init__(
         overwrite=False,
         bogus_last_correction=False,
     ):
-        if aliases is None:
-            aliases = []
-        for code in (tempo_code, itoa_code):
-            if code is not None:
-                aliases.append(code)
 
-        super().__init__(
-            name,
-            aliases=aliases,
-            include_gps=include_gps,
-            include_bipm=include_bipm,
-            bipm_version=bipm_version,
-            overwrite=overwrite,
-        )
-        # ITRF coordinates are required
-        if itrf_xyz is None:
-            raise ValueError("ITRF coordinates not given for observatory '%s'" % name)
-
-        # Convert coords to standard format.  If no units are given, assume
-        # meters.
-        if not has_astropy_unit(itrf_xyz):
-            xyz = np.array(itrf_xyz) * u.m
-        else:
-            xyz = itrf_xyz.to(u.m)
-
-        # Check for correct array dims
-        if xyz.shape != (3,):
+        input = [
+            location is not None,
+            itrf_xyz is not None,
+            (lat is not None and lon is not None and height is not None),
+        ]
+        if sum(input) == 0:
             raise ValueError(
-                "Incorrect coordinate dimensions for observatory '%s'" % (name)
+                "EarthLocation, ITRF coordinates, or lat/lon/alt are required for observatory '%s'"
+                % name
             )
-
-        # Convert to astropy EarthLocation, ensuring use of ITRF geocentric coordinates
-        self._loc_itrf = EarthLocation.from_geocentric(*xyz)
+        if sum(input) > 1:
```

Comment:
This doesn't catch the case where they specify `itrf_xyz` and `lon` (but not `lat`).

Reply:
done.